lib/devise/strategies/jwt_authenticatable.rb

```diff
@@ -0,0 +1,48 @@
+# frozen_string_literal: true
+require 'devise/strategies/authenticatable'
+require 'jwt'
+
+module Devise
+  module Strategies
+    class JwtAuthenticatable < Authenticatable
+      def valid?
+        super || token # Why super? Should always return false
+      end
+
+      def authenticate!
+        Rails.logger.debug("Authenticating with JWT.")
+        return fail(:invalid_token) if @token.blank? # Is a more sophisticated check required here?
```

Comment:
Don't use `@token` since getter method for this lazy loading value is available (`token`)